README.md

```diff
@@ -2,727 +2,369 @@
 
-In the following example, the URL ```http://mbed.org/device/?code=07050200623B61125D5EF72A``` for the STMicro Nucleo F302R8 board contains the valid TargetID ```07050200623B61125D5EF72A```, which ```mbed-ls``` uses to detect the ```platform_name```. ```mbed-ls``` will then replace the invalid TargetID in ```usb-id``` with the value from ```mbed.htm```.
+As a normal function definition:
 
-```html
-<!-- mbed Microcontroller Website and Authentication Shortcut -->
-<!-- Version: 0200 Build: Aug 27 2014 13:29:28 -->
-<html>
-<head>
-<meta http-equiv="refresh" content="0; url=http://mbed.org/device/?code=07050200623B61125D5EF72A"/>
-<title>mbed Website Shortcut</title>
-</head>
-<body></body>
-</html>
-```
+```python
+def filter_func(mbed):
+    return m['platform_name'] == 'K64F'
 
-This is the result of ```mbedls``` listing the connected devices that we saw above:
+mbeds.list_mbeds(filter_function=filter_func)
 ```
-$ mbedls
-+---------------------+-------------------+-------------------+----------------------------------------+
-|platform_name        |mount_point        |serial_port        |target_id                               |
-+---------------------+-------------------+-------------------+----------------------------------------+
-|KL25Z                |I:                 |COM89              |02000203240881BBD9F47C43                |
-|LPC1768              |H:                 |COM77              |101000000000000000000002F7F18695        |
-|NUCLEO_F302R8        |G:                 |COM34              |07050200623B61125D5EF72A                |
-|NUCLEO_L152RE        |E:                 |COM9               |07100200860579FAB960EFD7                |
-|unknown              |F:                 |COM5               |A000000001                              |
-+---------------------+-------------------+-------------------+----------------------------------------+
+
+As a lambda function:
+
+```python
+platforms = mbeds.list_mbeds(filter_function=lambda m: m['platform_name'] == 'K64F')
 ```
 
-# Retarget mbed-ls autodetection results
+##### `unique_names`
 
-User can create file ```mbedls.json``` in given directory. ```mbedls.json``` file should contain JSON formatted data which will redefine mbed's parameters returned by mbed-ls. ```mbed-ls``` will automatically read ```mbedls.json``` file and alter auto-detection result.
-File should be placed in directory where we want to alter mbed-ls behavior.
+**Default:** `False`.
 
-* Note: This feature in implicitly ON.
-* Note: This feature can be turned off with command line switch ```--skip-retarget```.
+This controls whether a unique name is assigned to each platform. The unique name takes the form of `K64F[0]`, where the number between the brackets is an incrementing value. This name is accessible through the dictionary member `platform_unique_name` in the returned platform data. It defaults to `False`.
```

Comment:
Query: Who or what assigns a unique name to each platform? You, the user? Or something else? Please clarify for active voice.